File: app/src/main/java/com/ako/myexcel/DBHelper.java
package com.ako.myexcel;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteException;
import android.database.sqlite.SQLiteOpenHelper;
import android.util.Log;

import java.util.ArrayList;
import java.util.HashMap;



public class DBHelper {

    String TAG = "DBAdapter";

    public static final String Tablename = "MyTable1";
    public static final String id = "_id";// 0 integer
    public static final String Company = "Company";// 1 text(String)
    public static final String Product = "Product";// 2 integer
    public static final String Price = "Price";// 3 date(String)

    private SQLiteDatabase db;
    private Database dbHelper;

    private static final int VERSION = 1;
    private static final String DB_NAME = "MyDB1.db";

    public DBHelper(Context context) {
        dbHelper = new Database(context);
    }

    public void open() {
        if (null == db || !db.isOpen()) {
            try {
                db = dbHelper.getWritableDatabase();
            } catch (SQLiteException sqLiteException) {
            }
        }
    }

    public void close() {
        if (db != null) {
            db.close();
        }
    }

    public int insert(String table, ContentValues values) {
        try {
            db = dbHelper.getWritableDatabase();
            int y = (int) db.insert(table, null, values);
            db.close();
            Log.e("Data Inserted", "Data Inserted");
            Log.e("y", y + "");
            return y;
        } catch (Exception ex) {
            Log.e("Error Insert", ex.getMessage().toString());
            return 0;
        }
    }

    public void delete() {
        db.execSQL("delete from " + Tablename);
    }

    public Cursor getAllRow(String table) {
        return db.query(table, null, null, null, null, null, id);
    }

    public ArrayList<HashMap<String, String>> getProducts() {
        ArrayList<HashMap<String, String>> prolist;
        prolist = new ArrayList<HashMap<String, String>>();
        String selectQuery = "SELECT * FROM " + Tablename;
        SQLiteDatabase database = dbHelper.getReadableDatabase();
        Cursor cursor = database.rawQuery(selectQuery, null);
        if (cursor.moveToFirst()) {
            do {
                HashMap<String, String> map = new HashMap<String, String>();
                map.put(Company, cursor.getString(1));
                map.put(Product, cursor.getString(2));
                map.put(Price, cursor.getString(3));
                prolist.add(map);
            } while (cursor.moveToNext());
        }
        return prolist;
    }

    private class Database extends SQLiteOpenHelper {
        private static final int VERSION = 1;
        private static final String DB_NAME = "MyDB1.db";

        public Database(Context context) {
            super(context, DB_NAME, null, VERSION);
        }

        @Override
        public void onCreate(SQLiteDatabase db) {
            String create_sql = "CREATE TABLE IF NOT EXISTS " + Tablename + "("
                    + id + " INTEGER PRIMARY KEY AUTOINCREMENT,"
                    + Company + " TEXT ," + Product + " TEXT ,"
                    + Price + " TEXT " + ")";
            db.execSQL(create_sql);
        }

        @Override
        public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
            db.execSQL("DROP TABLE IF EXISTS " + Tablename);
        }

    }
}

File: app/src/main/java/com/ako/myexcel/DBHelper.kt
package com.ako.myexcel

import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteException
import android.database.sqlite.SQLiteOpenHelper
import android.util.Log

class DBHelper(context: Context?) {
    var TAG = "DBAdapter"
    private var db: SQLiteDatabase? = null
    private val dbHelper: Database

    init {
        dbHelper = Database(context)
    }

    fun open() {
        if (null == db || !db!!.isOpen) {
            try {
                db = dbHelper.writableDatabase
            } catch (sqLiteException: SQLiteException) {
            }
        }
    }

    fun close() {
        if (db != null) {
            db!!.close()
        }
    }

    fun insert(table: String?, values: ContentValues?): Int {
        return try {
            db = dbHelper.writableDatabase
            db?.insert(table, null, values)
            db?.close()
            Log.e("Data Inserted", "Data Inserted")
            1
        } catch (ex: Exception) {
            Log.e("Error Insert", ex.message.toString())
            0
        }
    }

    fun delete() {
        db!!.execSQL("delete from " + Tablename)
    }

    fun getAllRow(table: String?): Cursor {
        return db!!.query(table, null, null, null, null, null, id)
    }

    val products: ArrayList<HashMap<String, String>>
        get() {
            val prolist: ArrayList<HashMap<String, String>>
            prolist = ArrayList()
            val selectQuery = "SELECT * FROM " + Tablename
            val database = dbHelper.readableDatabase
            val cursor = database.rawQuery(selectQuery, null)
            if (cursor.moveToFirst()) {
                do {
                    val map = HashMap<String, String>()
                    map[company] = cursor.getString(1)
                    map[product] = cursor.getString(2)
                    map[price] = cursor.getString(3)
                    prolist.add(map)
                } while (cursor.moveToNext())
            }
            return prolist
        }
    private inner class Database(context: Context?) :
        SQLiteOpenHelper(context, "MyDB1.db", null, 1) {
        override fun onCreate(db: SQLiteDatabase) {
            val create_sql = ("CREATE TABLE IF NOT EXISTS " + Tablename + "("
                    + id + " INTEGER PRIMARY KEY AUTOINCREMENT,"
                    + company + " TEXT ," + product + " TEXT ,"
                    + price + " TEXT " + ")")
            db.execSQL(create_sql)
        }

        override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
            db.execSQL("DROP TABLE IF EXISTS " + Tablename)
        }
    }

    companion object {
        const val Tablename = "MyTable1"
        const val id = "_id" // 0 integer
        const val company = "Company" // 1 text(String)
        const val product = "Product" // 2 integer
        const val price = "Price" // 3 date(String)
        private const val VERSION = 2
        private const val DB_NAME = "MyDB1.db"
    }
}